.github/workflows/ci.yml

```diff
@@ -13,9 +13,59 @@ env:
   API_CLIENT_SECRET: clientsecret
 
 jobs:
-  test:
+  type_checking:
+    name: "Type check 🔎"
     runs-on: ubuntu-latest
+    steps: 
+      - name: Check out code
+        uses: actions/checkout@v4.1.7
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v4.0.3
+        with:
+          node-version-file: '.node-version'
+          cache: 'npm'
+
+      - name: Installing dependencies
+        run: npm install       
+
+      - name: Building source
+        run: npm run build
```

Comment:
I don't think we need to run the full build to check linting or typechecking? This should speed things up yet a bit more (albeit for the fastest jobs anyway!)